sdk/core/src/client_request_id.rs

```diff
@@ -0,0 +1,29 @@
+use crate::headers::*;
+use crate::AddAsHeader;
+use http::request::Builder;
+use std::borrow::Cow;
+
+#[derive(Debug, Clone)]
+pub struct ClientRequestId<'a>(Cow<'a, str>);
+
+impl<'a> ClientRequestId<'a> {
+    pub fn new<CRI: Into<Cow<'a, str>>>(client_request_id: CRI) -> Self {
+        Self(client_request_id.into())
+    }
+}
+
+impl<'a> AddAsHeader for ClientRequestId<'a> {
+    fn add_as_header(&self, builder: Builder) -> Builder {
+        builder.header(CLIENT_REQUEST_ID, self.0.as_ref())
+    }
+}
+
+impl<'a> AddAsHeader for Option<ClientRequestId<'a>> {
```

Comment:
Is this really needed when we'll be using the `add_header` helper function?

Reply:
yes it's probably best to the helper function 👍 !